src/pages/spawn/spawnVolume/spawnVolumeTable/spawnVolumeTableActions/migrateBtn/MigrateVolumeModal.tsx

```diff
@@ -0,0 +1,166 @@
+    MigrateVolumeMutation,
+    MigrateVolumeMutationVariables
+  >(MIGRATE_VOLUME, {
+    onCompleted() {
+      dispatchToast.success(
+        "Volume migration has been scheduled. A new host will be spawned and accessible on your Hosts page."
+      );
+      setOpen(false);
+    },
+    onError(err) {
+      dispatchToast.error(
+        `There was an error during volume migration: ${err.message}`
+      );
+      setSubmitClickCount(0);
+    },
+    refetchQueries: ["MyHosts", "MyVolumes", "GetMyPublicKeys"],
+  });
+  const [formState, setFormState] = useState<FormState>({});
+
+  const distros = useMemo(
+    () => formSchemaInput.distros?.filter((d) => d.isVirtualWorkStation),
+    [formSchemaInput.distros]
+  );
+  const { schema, uiSchema } = getFormSchema({
+    ...formSchemaInput,
+    distros,
+    isMigration: true,
+    isVirtualWorkstation: !!formState?.distro?.isVirtualWorkstation,
+  });
+  useVirtualWorkstationDefaultExpiration({
+    disableExpirationCheckbox: formSchemaInput.disableExpirationCheckbox,
+    formState,
+    setFormState,
+  });
+
+  useEffect(() => {
+    if (!open) {
+      setFormState({});
+    }
+  }, [open]);
+
+  const migrateVolume = useCallback(() => {
+    const mutationInput = formToGql({
+      formData: formState,
+      myPublicKeys: formSchemaInput.myPublicKeys,
+      migrateVolumeId: volume.id,
+    });
+    sendEvent({
+      name: "Spawned a host",
+      isMigration: true,
+      params: omit(mutationInput, [
+        "publicKey",
+        "userDataScript",
+        "setUpScript",
+      ]),
+    });
+    migrateVolumeMutation({
+      variables: {
+        spawnHostInput: mutationInput,
+        volumeId: volume.id,
+      },
+    });
+  }, [
+    formSchemaInput.myPublicKeys,
+    formState,
+    volume,
+    migrateVolumeMutation,
+    sendEvent,
+  ]);
+
+  useEffect(() => {
+    if (submitClickCount === 2) {
+      migrateVolume();
+    }
+  }, [setSubmitClickCount, migrateVolume, submitClickCount]);
+
+  if (loadingFormData) {
+    return null;
+  }
+  const title =
+    submitClickCount === 0
+      ? "Migrate Volume"
+      : "Are you sure you want to migrate this home volume?";
+
+  let buttonText = "Migrate Volume";
+  if (loadingMigration) {
+    buttonText = "Migrating";
+  } else if (submitClickCount === 0) {
+    buttonText = "Next";
+  }
+
+  return (
+    <ConfirmationModal
+      title={title}
+      open={open}
+      submitDisabled={
+        !validateSpawnHostForm(formState, true) || loadingMigration
+      }
+      onConfirm={() => setSubmitClickCount(submitClickCount + 1)}
+      data-cy="migrate-modal"
+      buttonText={buttonText}
+      onCancel={
+        submitClickCount === 0
+          ? () => setOpen(false)
+          : () => setSubmitClickCount(0)
+      }
+    >
+      <Body>
+        Migrate this home volume to a new host. Upon successful migration, the
+        unused host will be scheduled to expire in 24 hours.
+      </Body>
```

Comment:
Is this supposed to show up twice? 
<img width="603" alt="image" src="https://user-images.githubusercontent.com/4605522/199352876-85592cc5-391c-4523-a06e-04144c22c8cf.png">

<img width="748" alt="image" src="https://user-images.githubusercontent.com/4605522/199352898-a49e1c1a-2574-42b3-8a83-4c6171a843af.png">